components/header/navbar/searchpage/index.tsx

```diff
@@ -0,0 +1,292 @@
+import { titleToAnchor } from '@utils/helpers'
+import ListCard from './ListCard'
+import SearchBar from './SearchBar'
+import { MdCancel } from 'react-icons/md'
+import { drive_v3 } from 'googleapis'
+import FileCard from './FileCard'
+
+type SearchOverlayProps = {
+  onClose: () => void
+  sessionToken?: string
+  setSideMenuOpen: (state: boolean) => void
+}
+
+const SearchOverlay = ({
+  onClose,
+  sessionToken,
+  setSideMenuOpen,
+}: SearchOverlayProps) => {
+  const [query, setQuery] = useState('')
+  const [fileSearch, setFileSearch] = useState(false)
+  const [searching, setSearching] = useState(false)
+  const [isFetched, setIsFetched] = useState(false)
+  const [fileResults, setFileResults] = useState<drive_v3.Schema$FileList>({
+    files: [],
+  })
+  const [results, setResults] = useState<SearchData>({
+    sectionData: [],
+    pageData: [],
+    privateSectionData: [],
+    privatePageData: [],
+  })
+
+  const clearResults = () => {
+    setResults({
+      sectionData: [],
+      pageData: [],
+      privateSectionData: [],
+      privatePageData: [],
+    })
+    setFileResults({ files: [] })
+    setIsFetched(false)
+  }
+
+  //search function for content or files depending on fileSearch state
+  //append is used to load more files with same query
+  const handleSearch = async (append = false) => {
+    //do not search if query is less than 2 characters
+    if (query.length < 2) return
+    try {
+      setSearching(true)
+      if (fileSearch) {
+        const resDrive = await searchDrive(
+          query,
+          //only nextPage if append is true
+          append ? fileResults.nextPageToken ?? undefined : undefined
+        )
+        //if append then add to existing files else reset
+        if (append) {
+          resDrive &&
+            append &&
+            setFileResults({
+              files: [...(fileResults.files ?? []), ...(resDrive.files || [])],
+              nextPageToken: resDrive.nextPageToken,
+            })
+        } else {
+          setFileResults(resDrive ?? { files: [] })
+        }
+        setSearching(false)
+      } else {
+        const res = await searchSiteContent(query, sessionToken)
+        setResults({
+          sectionData: res.sectionData,
+          pageData: res.pageData,
+          privateSectionData: res.privateSectionData,
+          privatePageData: res.privatePageData,
+        })
+        setSearching(false)
+      }
+      setIsFetched(true)
+    } catch (error) {
+      setSearching(false)
+      console.error(error)
+    }
+  }
+  // Add class to body to disable interaction with main page
+  useEffect(() => {
+    document.body.classList.add('overflow-hidden')
+    document.body.style.pointerEvents = 'none'
+    return () => {
+      document.body.classList.remove('overflow-hidden')
+      document.body.style.pointerEvents = 'auto'
+    }
+  }, [])
+
+  //search with new param when search content changes
+  useEffect(() => {
+    handleSearch()
+  }, [fileSearch])
```

Comment:
This should not probably be a `useEffect`, instead `handleSearch` should be called from the event handler